llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -951,6 +951,9 @@ bool AMDGPUTargetMachine::isNoopAddrSpaceCast(unsigned SrcAS,
 }
 
 unsigned AMDGPUTargetMachine::getAssumedAddrSpace(const Value *V) const {
+  if (isa<AllocaInst>(V))
```

Comment:
Is this only reachable for flat typed values?

Reply:
I don't think we need to differentiate them. It is either flat or private. It can't be something else, especially after https://github.com/llvm/llvm-project/pull/135820.